mw-webapp/src/component/header/Header.tsx

```diff
@@ -15,14 +12,21 @@ const LINK_TEXT = "Workflow";
 const LOGO_TEXT = "master's way";
 
 /**
- * Header component
+ * Header props
  */
-export const Header = () => {
-  const [user, setUser] = useState<User | null>(null);
+interface HeaderProps {
+
+  /**
+   * Current user
+   */
+  user: User | null;
+}
 
-  useEffect(() => {
-    handleUserAuthState(setUser);
-  }, []);
+/**
+ * Header component
+ */
+export const Header = (props: HeaderProps) => {
+  const user = props.user;
```

Comment:
1.  you can use destruction, it modern style: const {user} = props.
2. I believe that context means a lot, yeah sometimes we should destruct some variables for some reasons, fo example just to be short, or if it used in complex expression or in clean environment. In current case I dont see any reasons to destruct user, what do you think? If it is not clear we can discuss it on the nearest meeting, just remind me)

Reply:
fixed